src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.Fixer.cs

```diff
@@ -0,0 +1,180 @@
+                out bool isInvoked)
+        {
+            // The analyzer only passes a diagnostic for two scenarios, each having an IArrayCreationOperation:
+            //      1. The node is an IArgumentOperation that is a direct parent of an IArrayCreationOperation
+            //      2. The node is an IArrayCreationOperation already, as it was pulled from an
+            //         invocation, like with LINQ extension methods
+
+            // If this is a LINQ invocation, the node is already an IArrayCreationOperation
+            if (model.GetOperation(node, cancellationToken) is IArrayCreationOperation arrayCreation)
+            {
+                isInvoked = true;
+                return arrayCreation;
+            }
+            // Otherwise, we'll get the IArrayCreationOperation from the argument node's child
+            isInvoked = false;
+            return (IArrayCreationOperation)model.GetOperation(node.ChildNodes().First(), cancellationToken);
+        }
+
+        private static string GetExtractedMemberName(IEnumerable<string> memberNames, string parameterName)
+        {
+            bool hasCollectionEnding = s_collectionMemberEndings.Any(x => parameterName.EndsWith(x, true, CultureInfo.InvariantCulture));
+
+            if (parameterName == "source" // for LINQ, "sourceArray" is clearer than "source"
+                || (memberNames.Contains(parameterName) && !hasCollectionEnding))
+            {
+                parameterName += "Array";
+            }
+
+            if (memberNames.Contains(parameterName))
+            {
+                int suffix = 0;
+                while (memberNames.Contains(parameterName + suffix))
+                {
+                    suffix++;
+                }
+                return parameterName + suffix;
+            }
+
+            return parameterName;
+        }
+
+        private static string GetMemberNameFromType(IArrayCreationOperation arrayCreationOperation)
+        {
+#pragma warning disable CA1308 // Normalize strings to uppercase
+            return ((IArrayTypeSymbol)arrayCreationOperation.Type).ElementType.OriginalDefinition.Name.ToLowerInvariant() + "Array";
+#pragma warning restore CA1308 // Normalize strings to uppercase
+        }
+
+        private static Accessibility GetAccessibility(ISymbol? methodSymbol)
+        {
+            if (methodSymbol is not null)
+            {
+                // If private or public, return private since public accessibility not wanted for fields by default
+                return methodSymbol.GetResultantVisibility() switch
+                {
+                    // Return internal if internal
+                    SymbolVisibility.Internal => Accessibility.Internal,
+                    _ => Accessibility.Private
+                };
+            }
+            return Accessibility.Private;
```

Comment:
[minor] You can merge the if and the switch.

```suggestion
            // If private or public, return private since public accessibility not wanted for fields by default
            return methodSymbol?.GetResultantVisibility() switch
            {
                // Return internal if internal
                SymbolVisibility.Internal => Accessibility.Internal,
                _ => Accessibility.Private
            };
```